src/Features/CSharp/Portable/Completion/KeywordRecommenders/StackAllocKeywordRecommender.cs

```diff
@@ -17,52 +17,11 @@ public StackAllocKeywordRecommender()
 
         protected override bool IsValidContext(int position, CSharpSyntaxContext context, CancellationToken cancellationToken)
         {
-            var node = context.TargetToken.Parent;
-
-            // At start of a file
-            if (node == null)
-            {
-                return false;
-            }
-
-            // After a cast or parenthesized expression: (Span<int>)stackalloc
-            if (context.TargetToken.IsAfterPossibleCast())
-            {
-                node = node.Parent;
-            }
-
-            // Inside a conditional expression: value ? stackalloc : stackalloc
-            while (node.IsKind(SyntaxKind.ConditionalExpression) &&
-                (context.TargetToken.IsKind(SyntaxKind.QuestionToken, SyntaxKind.ColonToken) || context.TargetToken.IsAfterPossibleCast()))
-            {
-                node = node.Parent;
-            }
-
-            // assignment: x = stackalloc
-            if (node.IsKind(SyntaxKind.SimpleAssignmentExpression))
-            {
-                return node.Parent.IsKind(SyntaxKind.ExpressionStatement);
-            }
-
-            // declaration: var x = stackalloc
-            if (node.IsKind(SyntaxKind.EqualsValueClause))
-            {
-                node = node.Parent;
-
-                if (node.IsKind(SyntaxKind.VariableDeclarator))
-                {
-                    node = node.Parent;
-
-                    if (node.IsKind(SyntaxKind.VariableDeclaration))
-                    {
-                        node = node.Parent;
-
-                        return node.IsKind(SyntaxKind.LocalDeclarationStatement, SyntaxKind.ForStatement);
-                    }
-                }
-            }
-
-            return false;
+            // Beginning with C# 8.0, stackalloc expression can be used inside other expressions
+            // whenever a Span<T> or ReadOnlySpan<T> variable is allowed.
+            return (context.IsAnyExpressionContext && !context.IsConstantExpressionContext) ||
```

Comment:
mixing && ! || together is tougher to read.  there's nothing wrong with statements :)